### PR TITLE
Add black, isort, flake8, mypy, py.typed

### PR DESCRIPTION
**Pull request by atharvakale343**
Hi, I added some minor improvements. If you like them, you can choose to accept.

1. Adds a dev dependencies section to pyproject.toml

They can be installed using
```bash
pip install -e '.[dev]'
```

2. Adds an empty `py.typed` file to the root package

This advertises to package installers that this package is typed. This prevents mypy giving this error to package users:

```error
error: Skipping analyzing "flask_ml.flask_ml_server.models": module is installed, but missing library stubs or py.typed marker  [import-untyped]
```

3. Adds a Makefile that runs black, isort, flake8 and mypy

The Makefile can be simply run using:

```bash
make
```

black, isort, flake8 has not been applied in this PR to keep the diff short.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -25,5 +25,20 @@ dependencies = [
...
+    "flake8",
+    "mypy",
+    "types-requests",
+    "pytest"
+]
+
+[tool.setuptools.package-data]
+"flask_ml" = ["py.typed"]
+[tool.setuptools.packages.find]
+where = ["src"]
```
Comment (atharvakale343) [inline pyproject.toml:41]: This allows package installers know that this package is typed.

File: server_example.py [hunk 2]
```diff
@@ -46,18 +46,18 @@ def process_text(inputs: list[TextInput], parameters: dict):
 @server.route("/randomsentimentanalysis", DataTypes.TEXT)
 def sentiment_analysis(inputs: list[TextInput], parameters: dict):
     results = sentiment_model.predict(inputs)
-    results = [TextResult(text=res["text"], result=res["sentiment"]) for res in results]
-    response = ResponseModel(results=results)
+    text_results = [TextResult(text=res["text"], result=res["sentiment"]) for res in results]
+    response = ResponseModel(results=text_results)
```
Comment (atharvakale343) [inline server_example.py:50]: reassignment of variables is disallowed by `mypy` due to changing types

File: src/flask_ml/flask_ml_client/MLClient.py [hunk 3]
```diff
@@ -46,7 +48,7 @@ def request(
             return ErrorResponseModel(
                 status=f"Unknown error. status_code={str(response.status_code)}",
                 errors=[{"msg": UNKNOWN_ERROR}],
-            ).dict()
+            ).model_dump()
```
Comment (atharvakale343) [inline src/flask_ml/flask_ml_client/MLClient.py:51]: `dict()` is deprecated

File: src/flask_ml/flask_ml_server/models.py [hunk 4]
```diff
@@ -130,7 +130,7 @@ class ResponseModel(BaseModel):
         description="The status of the operation, e.g., 'SUCCESS'",
         min_length=1,
     )
-    results: List[Union[TextResult, ImageResult, AudioResult, VideoResult]] = Field(
+    results: Sequence[Union[TextResult, ImageResult, AudioResult, VideoResult]] = Field(
```
Comment (atharvakale343) [inline src/flask_ml/flask_ml_server/models.py:133]: `List` results in a `mypy` error. I think https://mypy.readthedocs.io/en/stable/common_issues.html#invariance-vs-covariance and https://blog.meadsteve.dev/programming/2023/09/09/typed-python-prefer-sequence-over-list/ somewhat explains it.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,7 @@
+SRC = $(wildcard *.py) $(shell find src tests -type f -name '*.py')
+
+format:
+	black --line-length 110 $(SRC)
+	isort --profile black $(SRC)
+	flake8 --max-line-length=110 --ignore=E203,W503 --select=F,N $(SRC)
+	mypy $(SRC)
```
Comment (prasannals) [inline Makefile:7]: We could consider having the mypy check in a separate make command. Not at issue for now.

**Comment by prasannals** [comment]
Thanks for this, Atharva. I don't have any major comments. This looks good. Merging it!

**Comment by prasannals** [comment]
I will merge this and fix the failing test. It's a minor fix. I think the syntax you used only works with Python 3.10 and above.

**Pull request merged**